independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java

```diff
@@ -361,7 +361,7 @@ public void handle(AsyncResult<Buffer> ar) {
                                         try {
                                             if (buffer.length() > 0) {
                                                 requestContext.setResponseEntityStream(
-                                                        new ByteArrayInputStream(buffer.getBytes()));
+                                                        new ByteBufInputStream(buffer.getByteBuf()));
```

Comment:
beware this ... the buffer is not releaseable?  when it's going to be used?

If is unreleaseable and perform an additional retain - it risks to cause a leak

Reply:
It seems fine since:
- https://netty.io/4.1/api/io/netty/buffer/ByteBuf.html#slice-- is called - and that doesn't increase the ref cnt
- https://github.com/eclipse-vertx/vert.x/blob/d49eb4caf16a50e619157d4343db872914b3bf7a/vertx-core/src/main/java/io/vertx/core/buffer/impl/BufferImpl.java#L556 is where we "should" go, but it's fine if we won't too, since
- https://github.com/netty/netty/blob/cd3dfe9b632671e7f8f7e5ca44918b34e888e14a/buffer/src/main/java/io/netty/buffer/ByteBufInputStream.java#L80 won't `release` it

So, the only thing we have to be sure is that it is used immediately and won't survive any subsequent `release` performed by Vertx/Netty which will release it whilst not yet used